libs/components/src/lib/Theme/index.hook.ts

```diff
@@ -0,0 +1,161 @@
+import { useCallback, useLayoutEffect, useState } from "react";
+// locals
+import { FontFamily, FontWeight, ThemeColor } from "./index.type";
+
+export const useData = () => {
+  const [themeColor, setThemeColor] = useState<ThemeColor>("blue");
+  const [fontFamily, setFontFamily] = useState<FontFamily>("sansSerif");
+  const [fontWeight, setFontWeight] = useState<FontWeight>("medium");
+
+  /* -------------------------- Update local storage -------------------------- */
+
+  const updateThemeColorLocalStorage = useCallback((value: ThemeColor) => {
+    localStorage.setItem("theme-color", value);
+  }, []);
+
+  const updateFontFamilyLocalStorage = useCallback((value: FontFamily) => {
+    localStorage.setItem("font-family", value);
+  }, []);
+
+  const updateFontWeightLocalStorage = useCallback((value: FontWeight) => {
+    localStorage.setItem("font-weight", value);
+  }, []);
+
+  /* ----------------------------- Update document ---------------------------- */
+
+  const updateThemeColorDocument = useCallback((value: ThemeColor) => {
+    let id = "";
+    const element = document.querySelector(
+      "[data-theme=theme-color]"
+    ) as HTMLElement;
+    if (!element) return;
+    switch (value) {
```

Comment:
instead switch case use an object and get id with object key like this:
 const colorValue: {[id: string]: ThemeColor} = {
    "blue": "blue",
    "green": "green",
    "purple": "purple",
    "gold": "gold",
    "grey": "grey",
  }

  const selectColor = (value: ThemeColor) => {
    return colorValue["blue"]
  }

Reply:
import { ThemeProvider } from "../src/lib/theme-provider";

export const WithMuiTheme = (Story, context) => {
  const { theme } = context.globals;

  return (
    <ThemeProvider themeMode={theme}>
      <Story />
    </ThemeProvider>
  );
};

export const decorators = [WithMuiTheme];

export const globalTypes = {
  theme: {
    name: "Theme",
    title: "Theme",
    description: "Theme for your components",
    defaultValue: "light",
    toolbar: {
      icon: "paintbrush",
      dynamicTitle: true,
      items: [
        { value: "light", left: "☀️", title: "Light mode" },
        { value: "dark", left: "🌙", title: "Dark mode" },
      ],
    },
  },
};